Update retention documentation for 2.0 (#502)
timescale/docs.timescale.com-content: using-timescaledb/data-retention.md

The scheduler framework also allows one to view scheduled jobs:
```sql

> SELECT * FROM timescaledb_information.policy_stats;

Reviewer: This will probably be renamed to job_stats but final decission is still pending

Reviewer 2: Update to `job_stats`

```

For more information, please see the [API documentation][add_retention_policy].

### Data Retention with Continuous Aggregates

> Extra care must be taken when using rention policies or `drop_chunks` calls on

Reviewer 2: ```suggestion
Extra care must be taken when using retention policies or `drop_chunks` calls on
```

hypertables which have [continuous aggregates][continuous_aggregates] defined on

> them. Any chunks which are dropped in the region still being refreshed by the

Reviewer 3: This is a defining clause.
```suggestion
them. Any chunks that are dropped in the region still being refreshed by the
```

> them. Any chunks which are dropped in the region still being refreshed by the

Reviewer 4: ```suggestion
them. Similar to a refresh of a materialized view, a refresh on a continuous aggregate will update the aggregate to reflect changes in the underlying source data. This means that any chunks which are dropped in the region still being refreshed by the
```

Author: I really liked how you captured this, added

continuous aggregate will cause the chunk data to disappear from the aggregate as
well. If the intent is to keep the aggregate while dropping the underlying data,
the interval being dropped should not overlap with the offsets for the continuous

> aggregate.  More detail on this can be found [here][cagg_drop_data].

Reviewer 3: ```suggestion
aggregate.  More detail on this can be found in the section [whatever the name of the section is][cagg_drop_data].
```

> One additional item to note is that continuous aggregates are also valid targets

Reviewer 3: ```suggestion
Also note that continuous aggregates are valid targets
```

for `drop_chunks` and retention policies.  As an example, assume you have your
`conditions` hypertable from above and a `conditions_summary_daily` continuous
aggregate which stores the average temperature per device per day.  The following
will drop the data from `conditions` after 30 days, but will keep the
`conditions_summary_daily` around for an additional 570 days.

```sql
SELECT add_retention_policy('conditions', INTERVAL '30 days');
SELECT add_retention_policy('conditions_summary_daily', INTERVAL '600 days');
```

> Note that for this to work properly, it is important that the `end_offset` for

Reviewer 3: ```suggestion
For this to work properly, it is important that the `end_offset` for
```

Reviewer 2: Should we show this explicitly in the example, then?  I'm trying to follow this, but with the changes I see proposed, I don't see the definition for conditions_summary_daily anywhere:

![image](https://user-images.githubusercontent.com/205086/94634671-a7e7b280-029e-11eb-83bc-f9ca3398bbb6.png)

Author: I've updated the example, hopefully it's much easier to follow now.

The scheduler framework also allows one to view scheduled jobs:
```sql
SELECT * FROM timescaledb_information.job_stats;
```

For more information, please see the [API documentation][add_retention_policy].

### Data Retention with Continuous Aggregates [](retention-with-aggregates)

Extra care must be taken when using retention policies or `drop_chunks` calls on
hypertables which have [continuous aggregates][continuous_aggregates] defined on
them. Similar to a refresh of a materialized view, a refresh on a continuous aggregate
will update the aggregate to reflect changes in the underlying source data. This means
that any chunks that are dropped in the region still being refreshed by the
continuous aggregate will cause the chunk data to disappear from the aggregate as
well. If the intent is to keep the aggregate while dropping the underlying data,
the interval being dropped should not overlap with the offsets for the continuous
aggregate.

As an example, let's add a continuous aggregate to our `conditions` hypertable:
```sql

> CREATE MATERIALIZED VIEW conditions_summary_daily (day, device, temp)
> WITH (timescaledb.continuous) AS
> SELECT time_bucket('1 day', time), device, AVG(temperature)
> FROM conditions
> GROUP BY (1, 2);

Reviewer 2: ```suggestion
CREATE MATERIALIZED VIEW conditions_summary_daily (day, device, temp)
WITH (timescaledb.continuous) AS
  SELECT time_bucket('1 day', time), device, avg(temperature)
  FROM conditions
  GROUP BY (1, 2);
```

Author: fixed

SELECT add_continuous_aggregate_policy('conditions_summary_daily', '7 days', '1 day', '1 day');
```

This will create the `conditions_summary_daily` aggregate which will store the daily
temperature per device from our `conditions` table. However, we have a problem here

> if we're using our 24 hour rentention policy from above, as our aggregate will capture

Reviewer 2: ```suggestion
if we're using our 24 hour retention policy from above, as our aggregate will capture
```

Author: Fixed

changes to the data for up to seven days. As a result, we will update the aggregate
when we drop the chunk from the table, and we'll ultimately end up with no data in our

> `conditions_summary_daily` table.

Reviewer 3: Hmmm... usually it is not recommended to show incorrect examples since misunderstanding are quite common. It is usually more common to show a correct example and then digress into the problematic case: along the lines "Note that ... if we did it this way instead, it would not work since...". Then you keep the "faulty" code located to the discussion about it.
